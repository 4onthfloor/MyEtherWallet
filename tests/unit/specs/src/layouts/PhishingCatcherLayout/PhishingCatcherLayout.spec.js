import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import PhishingCatcherLayout from '@/layouts/PhishingCatcherLayout/PhishingCatcherLayout.vue';
import { Tooling } from '@@/helpers';

describe('PhishingCatcherLayout.vue', () => {
  let localVue, i18n, wrapper, store;

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;
  });

  beforeEach(() => {
    wrapper = shallowMount(PhishingCatcherLayout, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      stubs: {
        'router-link': RouterLinkStub
      },
      computed: {
        linkQuery() {
          return {
            url: 'https://www.some-phishing-site.net',
            'phishing-address': 'https://www.some-phishing-site.net'
          };
        }
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Has url in linkQuery', () => {
    const url = 'https://www.some-phishing-site.net';
    const code = wrapper.find('code');
    const supposedValue = `<code>
          ${url}
        </code>`;
    expect(code.html()).toEqual(supposedValue);
  });

  it('dismounts', () => {
    wrapper.destroy();
    expect(wrapper.exists()).toBe(false);
  });
});
