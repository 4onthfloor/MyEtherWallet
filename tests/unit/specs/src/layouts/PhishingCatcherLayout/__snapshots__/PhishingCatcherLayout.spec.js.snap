// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhishingCatcherLayout.vue renders correctly 1`] = `
<div class="phishing-catcher-contianer">
  <div class="catcher-text">
    <h4>YOU HAVE BEEN REDIRECTED!</h4>
    <div>
      <div class="code-container"><code>
          https://www.some-phishing-site.net
        </code></div>
      <p>
        This site has been flagged for suspicious activity, <br>
        so we've redirected your browser. Stay safe out there
      </p>
    </div>
  </div>
</div>
`;
